tests/basic/pool_function_test.go

```diff
@@ -533,5 +520,84 @@ var _ = Describe("{PoolExpandWhileResizeDiskInProgress}", func() {
 		})
 
 	})
+})
+
+var _ = Describe("{PoolExpandTestLimits}", func() {
+	BeforeEach(func() {
+		contexts = scheduleApps()
+	})
+
+	JustBeforeEach(func() {
+		poolIDToResize = pickPoolToResize()
+		log.Infof("Picked pool %s to resize", poolIDToResize)
+		poolToResize = getStoragePool(poolIDToResize)
+	})
+
+	JustAfterEach(func() {
+		AfterEachTest(contexts)
+	})
+
+	AfterEach(func() {
+		appsValidateAndDestroy(contexts)
+		EndTorpedoTest()
+	})
+
+	It("Initiate pool expansion (DMThin) to its limits (15 TiB) and beyond", func() {
+		Step("Expand pool to 15 TiB (max supported capacity for DMThin) with resize-disk type. ", func() {
+			var testrailID = 51292
+			// testrailID corresponds to: https://portworx.testrail.net/index.php?/cases/view/51292
+
+			StartTorpedoTest("PoolExpandTestWithin15TiBLimit",
+				"Initiate pool expansion using resize-disk to 15 TiB target size", nil, testrailID)
+
+			// To achieve total pool size of 15 TiB:
+			// 1. Add another drive of same size for pool to have 2 drives.
+			// 2. Perform resize-disk operation which is faster than pool rebalance
+			//    due to adding a new 7 TiB drive.
+			targetSizeGiB := (poolToResize.TotalSize / units.GiB) * 2
+
+			log.InfoD("Next trying to expand the pool %s to %v GiB with type add-disk",
+				poolIDToResize, targetSizeGiB)
+			triggerPoolExpansion(poolIDToResize, targetSizeGiB, api.SdkStoragePool_RESIZE_TYPE_ADD_DISK)
+			resizeErr := waitForOngoingPoolExpansionToComplete(poolIDToResize)
+			dash.VerifyFatal(resizeErr, nil, "Pool expansion should not result in error")
+			verifyPoolSizeEqualOrLargerThanExpected(poolIDToResize, targetSizeGiB)
+
+			targetSizeTiB := uint64(15)
+			targetSizeInBytes = targetSizeTiB * units.TiB
+			targetSizeGiB = targetSizeInBytes / units.GiB
+
+			log.InfoD("Current Size of the pool %s is %d GiB. Trying to expand to %v TiB with type resize-disk",
+				poolIDToResize, targetSizeGiB, targetSizeTiB)
+			triggerPoolExpansion(poolIDToResize, targetSizeGiB, api.SdkStoragePool_RESIZE_TYPE_RESIZE_DISK)
+			resizeErr = waitForOngoingPoolExpansionToComplete(poolIDToResize)
+			dash.VerifyFatal(resizeErr, nil, "Pool expansion should not result in error")
+			verifyPoolSizeEqualOrLargerThanExpected(poolIDToResize, targetSizeGiB)
+
+		})
+
+		Step("Expand pool to 20 TiB (beyond max supported capacity for DMThin) with add-disk type. ", func() {
```

Comment:
We should check px install type here and only proceed to verify error occurs on expanding beyond 20TB if it is dmthin.

Reply:
Added check that only if PX is installed with DMThin, only then run this test otherwise skip.